storybook/utils/zeroReports.test.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * Zero Reports utility tests.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { replaceValuesOrRemoveRowForDateRangeInAnalyticsReport } from './zeroReports';
+import mockedReportResponse from '../../assets/js/modules/analytics-4/utils/__fixtures__/mocked-report-multiple-distinct-date-ranges.json';
+
+describe( 'zeroReports', () => {
+	describe( 'replaceValuesOrRemoveRowForDateRangeInAnalyticsReport', () => {
+		it( 'should zero values for date_range_0 when emptyRowBehavior is "zero"', () => {
+			// Create a deep copy of the mocked report
+			const report = JSON.parse( JSON.stringify( mockedReportResponse ) );
+			const result =
+				replaceValuesOrRemoveRowForDateRangeInAnalyticsReport(
+					report,
+					'date_range_0',
+					'zero'
+				);
+
+			// Expect all rows to be present (10 total)
+			expect( result.rows ).toHaveLength( 10 );
+
+			// Check that date_range_0 rows have zeroed metric values
+			const dateRange0Rows = result.rows.filter( ( row ) =>
+				row.dimensionValues.some(
+					( value ) => value.value === 'date_range_0'
+				)
```

Comment:
We have several repetitions of the same check/logic, that we can simplify by defining a util function also:
```js
const matchesDateRange = ( data, dateRange = 'date_range_0' ) => data.some( ( value ) => value.value === dateRange );
```

And then use it here and in other places in this file:
```suggestion
				matchesDateRange( row.dimensionValues )
```
And we would only need to pass additional parameters for date range where we check agains `date_range_1 `